torch/csrc/Device.cpp

```diff
@@ -137,6 +137,37 @@ PyObject *THPDevice_rc(PyObject *a, PyObject *b, int op) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPDevice_reduce(THPDevice *self)
+{
+  PyObject *ret, *mod, *obj;
```

Comment:
Please use `THPObjectPtr`s for those. They will automatically decref when they are destructed, so you won'thave to remember about it.

Reply:
It might also be easier to use `pybind`. See [this](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/script/init.cpp#L27-L28) for an example.